api/src/main/java/net/kyori/adventure/text/ComponentCompaction.java

```diff
@@ -166,6 +190,41 @@ static Component compact(final @NotNull Component self, final @Nullable Style pa
     return builder.build();
   }
 
+  /**
+  * Checks whether the Component is blank (a TextComponent containing only space characters)
+  *
+  * @param component the component to check
+  * @return true if the provided component is blank, false otherwise
+  */
+  private static boolean isBlank(Component component) {
+    if (component instanceof TextComponent) {
+      final TextComponent textComponent = (TextComponent) component;
+      
+      for (char c : textComponent.content().toCharArray()) {
```

Comment:
can you do this using indexes and `charAt`, to avoid allocating the character array?

Reply:
See 66d2ac3